rotkehlchen/api/rest.py

```diff
@@ -2929,147 +2932,151 @@ def refresh_transactions(
         return {'result': result, 'message': message, 'status_code': status_code}
 
     @async_api_call()
-    def decode_given_evm_transactions(
+    def decode_given_transactions(
             self,
-            transactions: list[tuple[SUPPORTED_CHAIN_IDS, EVMTxHash]],
+            transactions: list[tuple[CHAINS_WITH_TX_DECODING_TYPE, EVMTxHash | Signature]],
             delete_custom: bool,
     ) -> dict[str, Any]:
-        """
-        Repull data for the given transactions and redecode all events. Also prices for
-        the assets involved in these events are requeried.
-        """
-        task_manager = self.rotkehlchen.task_manager
-        assert task_manager, 'task manager should have been initialized at this point'
+        """Repull data for the given transactions and redecode all events."""
         success, message, status_code, events = True, '', HTTPStatus.OK, []
-        for evm_chain, tx_hash in transactions:
-            chain_manager = self.rotkehlchen.chains_aggregator.get_evm_manager(evm_chain)
-            with self.rotkehlchen.data.db.user_write() as write_cursor:
-                write_cursor.execute(
-                    'DELETE FROM evm_transactions WHERE tx_hash=? AND chain_id=?',
-                    (tx_hash, evm_chain.serialize_for_db()))
-            try:
-                chain_manager.transactions.get_or_query_transaction_receipt(tx_hash=tx_hash)
-            except RemoteError as e:
-                return {
-                    'result': False,
-                    'message': f'Failed to request evm transaction decoding due to hash {tx_hash.hex()} does not correspond to a transaction at {evm_chain.name}. {e!s}',  # noqa: E501
-                    'status_code': HTTPStatus.CONFLICT,
-                }
-            except DeserializationError as e:
-                return {
-                    'result': False,
-                    'message': f'Failed to request evm transaction decoding due to {e!s}',
-                    'status_code': HTTPStatus.CONFLICT,
-                }
+        for chain, tx_ref in transactions:
```

Comment:
Would it not make more sense to disallow combining a query of different types of chains in the same api call? Then you can just have a single if conditional at first instead of looping it and even organize it better in functions.

If so you would need to update docstring also.